File: Src/WitsmlExplorer.Api/Models/ChangeLog.cs
namespace WitsmlExplorer.Api.Models
{
    // ChangeLog extends ObjectOnWellbore despite not having uid and name fields in WITSML
    // These fields are set in the API to conform to ObjectOnWellbore
    public class ChangeLog : ObjectOnWellbore
    {
        public string UidObject { get; set; }
        public string NameObject { get; set; }
        public string LastChangeType { get; set; }
        public CommonData CommonData { get; set; }
    }
}

File: Src/WitsmlExplorer.Frontend/models/changeLog.ts
import CommonData from "./commonData";
import ObjectOnWellbore from "./objectOnWellbore";

// ChangeLog extends ObjectOnWellbore despite not having uid and name fields in WITSML
// These fields are set in the API to conform to ObjectOnWellbore
export default interface ChangeLog extends ObjectOnWellbore {
  uidObject: string;
  nameObject: string;
  lastChangeType: string;
  commonData: CommonData;
}
